ovs/lib/vdisk.py

```diff
@@ -617,14 +627,18 @@ def create_new(volume_name, volume_size, storagedriver_guid):
             raise
 
         with volatile_mutex('voldrv_event_disk_{0}'.format(volume_id), wait=30):
-            new_vdisk = VDisk()
-            new_vdisk.size = volume_size
-            new_vdisk.vpool = vpool
-            new_vdisk.devicename = devicename
-            new_vdisk.name = volume_name
-            new_vdisk.description = volume_name
-            new_vdisk.volume_id = volume_id
-            new_vdisk.save()
+            new_vdisk = VDiskList.get_vdisk_by_volume_id(volume_id)
+            if new_vdisk is None:
+                new_vdisk = VDiskList.get_by_devicename_and_vpool(devicename, vpool)
+                if new_vdisk is None:
+                    new_vdisk = VDisk()
+                    new_vdisk.size = volume_size
+                    new_vdisk.vpool = vpool
+                    new_vdisk.devicename = devicename
+                    new_vdisk.name = volume_name
```

Comment:
No, these ```extract_volumename``` code was actually a copy-paste leftover from the ```resize_from_voldrv``` call where - if the disk does not exist yet - it sets a best-guess name.